services/harmony/test/models/label.ts

```diff
@@ -0,0 +1,45 @@
+
+import { describe, it } from 'mocha';
+import { expect } from 'chai';
+import { buildJob, getFirstJob } from '../helpers/jobs';
+import { hookTransactionEach } from '../helpers/db';
+import { setLabelsForJob } from '../../app/models/label';
+
+describe('label CRUD', function () {
+  hookTransactionEach();
+  beforeEach(async function () {
+    this.job = await buildJob({ username: 'dummy' });
+    this.job.save(this.trx);
+  });
+
+  const labels = ['foo', 'Bar'];
+
+  describe('set labels for job', async function () {
+    it('sets the labels for the job', async function () {
+      await setLabelsForJob(this.trx, this.job.jobID, this.job.username, labels);
+      const newJob = await getFirstJob(this.trx);
+      expect(newJob.labels).deep.equal(labels.map((label) => label.toLowerCase()));
+    });
+  });
+
+  describe('update labels for job', async function () {
+    const updatedLabels = ['baz', 'buzz'];
+    it('updates the labels for the job', async function () {
+      await setLabelsForJob(this.trx, this.job.jobID, this.job.username, labels);
```

Comment:
nit: We can remove this line if we make the whole test a transaction with three steps and I think the test will be more readable if we do that.

Reply:
I like for tests to not rely on other tests setting things up in the database. It makes them too fragile.